guides/common/modules/proc_upgrading-el7-to-el8.adoc

```diff
@@ -0,0 +1,184 @@
+[id="Upgrading-Project-from-EL7-to-EL8{context}"]
+= Upgrading {Project} from {RHEL} 7 to {RHEL} 8
+
+Use this procedure to upgrade your {Project} installation from {RHEL} 7 to {RHEL} 8.
+LEAPP is the supported tool for {RHEL} 7 to {RHEL} 8 upgrades.
+
+.Prerequisites
+* {Project} {ProjectVersionPrevious} running on CentOS Linux 7.
+ifdef::katello,foreman-el[]
+- {Project} installations running on {RHEL} 7 cannot upgrade to {RHEL} 8. You must have a {RHEL} compatible LEAPP build to upgrade.
+endif::[]
+* Access to available repositories or a local mirror of repositories.
+* No outstanding updates.
+* {Project} installations can be upgraded to CentOS Stream 8 and a {RHEL} rebuild.
+ifdef::satellite[]
+* {Project} installations upgrade from {RHEL} 7 to {RHEL} 8, not from CentOS Linux 7 and {RHEL} rebuilds.
+endif::[]
+
+.Procedure
+ifdef::katello,foreman-el[]
+. Configure the https://copr.fedorainfracloud.org/coprs/g/theforeman/leapp/[@theforeman/leapp COPR Repository], which contains newer LEAPP packages than those shipped by https://wiki.almalinux.org/elevate/[AlmaLinux/ELevate], and support {Project} upgrades:
+
++
+----
+# curl -o /etc/yum.repos.d/theforeman-leapp.repo https://copr.fedorainfracloud.org/coprs/g/theforeman/leapp/repo/epel-7/group_theforeman-leapp-epel-7.repo
+----
+. Install required packages:
++
+----
+# yum install leapp leapp-repository leapp-data-centos
+----
+
+. Add {Project} specific repositories to `/etc/leapp/files/leapp_upgrade_repositories.repo`:
++
+----
+# [leapp-foreman]
+name=Foreman 3.2
+baseurl=https://yum.theforeman.org/releases/3.2/el8/$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+module_hotfixes=1
+
+[leapp-foreman-plugins]
+name=Foreman plugins 3.2
+baseurl=https://yum.theforeman.org/plugins/3.2/el8/$basearch
+enabled=1
+gpgcheck=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+module_hotfixes=1
+
+[leapp-foreman-client]
+name=Foreman client 3.2
+baseurl=https://yum.theforeman.org/client/3.2/el8/$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman-client
+
+[leapp-puppet7]
+name=Puppet 7 Repository el 8 - $basearch
+baseurl=http://yum.puppetlabs.com/puppet7/el/8/$basearch
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-puppet7-release
+       file:///etc/pki/rpm-gpg/RPM-GPG-KEY-2025-04-06-puppet7-release
+enabled=1
+gpgcheck=1
+
+[leapp-katello]
```

Comment:
can we use `ifdef::katello` here inside the code block? so we hide the katello repos in the foreman docs?

allows to drop the silly "If you are not using " comment below

Reply:
I used the following, and it worked:

```patch
diff --git guides/common/modules/proc_upgrading-el7-to-el8.adoc guides/common/modules/proc_upgrading-el7-to-el8.adoc
index ab89ff26..541e4aec 100644
--- guides/common/modules/proc_upgrading-el7-to-el8.adoc
+++ guides/common/modules/proc_upgrading-el7-to-el8.adoc
@@ -33,7 +33,7 @@ ifdef::katello,foreman-el[]
 . Add {Project} specific repositories to `/etc/leapp/files/leapp_upgrade_repositories.repo`:
 +
 ----
-# [leapp-foreman]
+[leapp-foreman]
 name=Foreman 3.2
 baseurl=https://yum.theforeman.org/releases/3.2/el8/$basearch
 enabled=1
@@ -64,6 +64,7 @@ gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-puppet7-release
 enabled=1
 gpgcheck=1
 
+ifdef::katello[]
 [leapp-katello]
 name=Katello 4.4
 baseurl=https://yum.theforeman.org/katello/4.4/katello/el8/$basearch/
@@ -87,11 +88,10 @@ gpgkey=https://yum.theforeman.org/pulpcore/3.16/GPG-RPM-KEY-pulpcore
 enabled=1
 gpgcheck=1
 module_hotfixes=1
+endif::[]
 ----
 endif::[]
 
-* If you are not using {Project}, you can leave out `leapp-katello`, `leapp-katello-candlepin` and `leapp-pulpcore`.
-
 * If you are using Puppet 6 instead of Puppet 7, replace the `7` with a `6` in the `leapp-puppet7` entry.
 
 * You need a Puppet repository for the Puppet agent that the installer is using.
```